### PR TITLE
added priorityQueue array

### PR DESCRIPTION
**Pull request by johannbarbie**

### CODE REVIEW

File: contracts/ParsecBridge.sol [hunk 1]
```diff
@@ -100,6 +110,15 @@ contract ParsecBridge is PriorityQueue {
     exitDuration = _exitDuration;
   }
 
+  function registerToken(ERC20 _token) public {
+    uint256[] memory arr = new uint256[](1);
```
Comment (Koroqe) [inline contracts/ParsecBridge.sol]: does `require(_psc != address(0));` needed?

File: contracts/ParsecBridge.sol [hunk 2]
```diff
@@ -72,15 +76,21 @@ contract ParsecBridge is PriorityQueue {
...
   }
   mapping(bytes32 => Exit) public exits;
 
 
-  constructor(ERC20 _token, uint256 _epochLength, uint256 _maxReward, uint256 _parentBlockInterval, uint256 _exitDuration) public {
-    // set token contract
-    require(_token != address(0));
-    token = _token;
+  constructor(ERC20 _psc, uint256 _epochLength, uint256 _maxReward, uint256 _parentBlockInterval, uint256 _exitDuration) public {
+    // set PSC contract
```
Comment (Koroqe) [inline contracts/ParsecBridge.sol]: mb just use `registerToken(_psc);`?

File: contracts/ParsecBridge.sol [hunk 3]
```diff
@@ -35,7 +36,9 @@ contract ParsecBridge is PriorityQueue {
   uint256 public averageGasPrice; // collected gas price for last submitted blocks
   uint256 exitDuration;
   bytes32 public tipHash; // hash of first period that has extended chain to some height
-  ERC20 public token;
+
+  mapping(uint256 => PriorityQueue.Token) public tokens;
```
Comment (Koroqe) [inline contracts/ParsecBridge.sol]: Should color be restricted to uint16 there?

File: contracts/ParsecBridge.sol [hunk 4]
```diff
@@ -454,15 +468,17 @@ contract ParsecBridge is PriorityQueue {
...
       height: periods[tipHash].height,
       owner: _owner,
+      color: _color,
       amount: _amount
     });
-    emit NewDeposit(depositCount, _owner, _amount);
+    emit NewDeposit(depositCount, _owner, _color, _amount);
   }
 
   function recoverTxSigner(uint256 offset, bytes32[] _proof) internal pure returns (address dest) {
```
[the remark this answers is not in the document]
Comment (johannbarbie) [reply contracts/ParsecBridge.sol]: more parameters, less compact in msg.data, i thought

File: contracts/ParsecBridge.sol (hunk 3 above)
Comment (johannbarbie) [reply contracts/ParsecBridge.sol]: maybe also for slots?

File: contracts/ParsecBridge.sol (hunk 2 above)
Comment (johannbarbie) [reply contracts/ParsecBridge.sol]: changed

**Pull request merged**